packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js

```diff
@@ -850,6 +852,84 @@ describe('#unit BoltProtocolV4x3', () => {
         expect(unpackedDateTimeWithoutOffset).toEqual(object)
       })
 
+      it('should pack and unpack DateTimeWithOffset', () => {
+        fc.assert(
+          fc.property(
+            fc.date({
+              min: temporalUtil.newDate(utils.MIN_UTC_IN_MS + utils.ONE_DAY_IN_MS),
+              max: temporalUtil.newDate(utils.MAX_UTC_IN_MS - utils.ONE_DAY_IN_MS)
+            }),
+            utils.arbitraryTimeZoneId(),
+            (date, timeZoneId) => {
+              const object = new DateTime(
+                date.getUTCFullYear(),
+                date.getUTCMonth() + 1,
+                date.getUTCDate(),
+                date.getUTCHours(),
+                date.getUTCMinutes(),
+                date.getUTCSeconds(),
+                date.getUTCMilliseconds() * 1_000_000,
+                undefined,
+                timeZoneId
+              )
+
+              buffer.reset()
+
+              const packable = protocol.packable(object)
+
+              expect(packable).not.toThrow()
+
+              buffer.reset()
+
+              const unpacked = protocol.unpack(buffer)
+
+              expect(unpacked.timeZoneOffsetSeconds).toBeDefined()
+
+              const unpackedDateTimeWithoutOffset = new DateTime(
+                unpacked.year,
+                unpacked.month,
+                unpacked.day,
+                unpacked.hour,
+                unpacked.minute,
+                unpacked.second,
+                unpacked.nanosecond,
+                undefined,
+                unpacked.timeZoneId
+              )
+
+              expect(loggerFunction)
```

Comment:
Is this expected to happen during packing or unpacking? As the test is right now, either would be accepted.

Reply:
During the packing, I will move the expectation to just after the packing.